diff --git a/hybris/common/hooks.c b/hybris/common/hooks.c
--- a/hybris/common/hooks.c
+++ b/hybris/common/hooks.c
@@ -450,9 +450,6 @@ static void* _hybris_hook_pthread_getspecific(pthread_key_t key)
 {
     TRACE_HOOK("key %d", key);
 
-    // see android_bionic/tests/pthread_test.cpp, test static_pthread_key_used_before_creation
-    if(!key) return NULL;
-
     return pthread_getspecific(key);
 }

